src/main/java/com/google/api/generator/gapic/composer/MethodSampleCodeHelperComposer.java

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer;
+
+import com.google.api.generator.engine.ast.AssignmentExpr;
+import com.google.api.generator.engine.ast.Expr;
+import com.google.api.generator.engine.ast.ExprStatement;
+import com.google.api.generator.engine.ast.MethodInvocationExpr;
+import com.google.api.generator.engine.ast.TryCatchStatement;
+import com.google.api.generator.engine.ast.TypeNode;
+import com.google.api.generator.engine.ast.Variable;
+import com.google.api.generator.engine.ast.VariableExpr;
+import com.google.api.generator.gapic.model.Method;
+import com.google.api.generator.gapic.model.MethodArgument;
+import com.google.api.generator.gapic.model.ResourceName;
+import com.google.api.generator.gapic.utils.JavaStyle;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class MethodSampleCodeHelperComposer {
+  private static String RESPONSE = "response";
+
+  public static TryCatchStatement composeUnaryRpcMethodSampleCode(
+      Method method,
+      List<MethodArgument> arguments,
+      TypeNode clientType,
+      Map<String, ResourceName> resourceNames) {
+    VariableExpr clientVarExpr = createVariableExpr(getClientName(clientType), clientType);
+    // Assign each method arguments with its default value.
+    Map<String, VariableExpr> methodArgVarExprMap = createMethodArgumentsVariableExprs(arguments);
```

Comment:
This seems to be doing a lot of extra work: (a) creating the VarExprs from the method args, (b) coupling the ordering of the VarExprs and MethodArguments to create AssignExprs, and then (c) gathering the VarExprs from the result. 

Instead, can we do the following, which would be much simpler:
- Add a VarExpr variant to DefaultValueComposer (and tests)
- Do (a), but in a one-liner not inside the wrapper method, and into a List
- Use the new method in DefaultValueComposer to do (b) directly, in a one-liner that's not wrapped in a method

Then the list of VarExprs can be passed directly into the method invocation builder.

Reply:
Not sure if I am understand correctly.
`Add a VarExpr variant to DefaultValueComposer (and tests)` -> if varExpr refer to a method argument variable expr, it only contains method argument's name and type, however to get a default value it require the method argument's `field`. I guess I didn't get the point of add VarExpr into DefaultValueComposer. Could you explain it more with details? Thanks

Reply:
PTAL at the `field` fields that DefaultValueComposer is actually using, and refactor them into separate methods (making any existing ones empty wrappers instead). For example:
- `private Expr createDefaultValue(TypeNode type, String varName)` would replace any non-`Field`-specific logic in the current `createDefaultValue(Field f)` method.
- `private Expr createDefaultValue(VariableExpr variableExpr)`
- `createDefaultValue(Field f)` would call the first one where possible.

Reply:
Instead of touching up the `DefaultValueComposer`, I propose other process:
1. create list of method argument variable, because the index of this list is same as list of a method arguments,
2. Use for-loop on the index to get default value expr and its variable expr. If the argument is String-type has reference name, we create new variable expr for it. And it is only place we need to use.
Please see `createMethodArgumentsVariableExprs` for more details.

Looking forward your suggestion on this. Thanks

Reply:
We'd like to avoid the index-based list coupling if we can. What is the motivation for going about it this way, rather than refactoring DefaultValueComposer?